src/mock_parsec/mod.rs

```diff
@@ -184,10 +184,21 @@ where
         unimplemented!()
     }
 
+    // TODO: rename this to `has_unpolled_observations`
```

Comment:
This isn't done yet because you would want to update Parsec to one that exposes the new API in the same PR?
I would add this to the comment.

Reply:
Yes, I'd like to make the change in Parsec first, to avoid the situation where routing becomes uncompilable with `--features=mock_base` (that is, with real parsec).

Reply:
Yes, I'd like to make the change in Parsec first, to avoid the situation where routing becomes uncompilable with `--features=mock_base` (that is, with real parsec).